pdi-dataservice-client/src/main/java/org/pentaho/di/core/sql/SQLCondition.java

```diff
@@ -372,28 +374,35 @@ private Condition parseAtomicCondition( String clause ) throws KettleSQLExceptio
       new int[] { Condition.FUNC_NOT_EQUAL, Condition.FUNC_LARGER_EQUAL, Condition.FUNC_LARGER_EQUAL,
         Condition.FUNC_SMALLER_EQUAL, Condition.FUNC_SMALLER_EQUAL, Condition.FUNC_SMALLER, Condition.FUNC_LARGER,
         Condition.FUNC_EQUAL, Condition.FUNC_REGEXP, Condition.FUNC_IN_LIST, Condition.FUNC_NOT_NULL,
-        Condition.FUNC_NULL, Condition.FUNC_LIKE, Condition.FUNC_CONTAINS, };
+        Condition.FUNC_NULL, Condition.FUNC_LIKE, Condition.FUNC_CONTAINS };
     int index = 0;
+
     while ( index < clause.length() ) {
       index = ThinUtil.skipChars( clause, index, '\'', '"' );
       for ( String operator : operators ) {
         if ( index <= clause.length() - operator.length() ) {
           if ( clause.substring( index ).toUpperCase().startsWith( operator ) ) {
             int functionIndex = Const.indexOfString( operator, operators );
+            boolean negation = false;
 
             // OK, we found an operator.
             // The part before is the first string
             //
             String left = Const.trim( clause.substring( 0, index ) );
             String op = Condition.functions[functions[functionIndex]];
             String right = Const.trim( clause.substring( index + operator.length() ) );
-            strings.add( left );
-            strings.add( op );
-            strings.add( right );
-            return strings;
+
+            if ( Pattern.matches( "^NULL$", right.toUpperCase() ) ) {
+              op = Condition.functions[Condition.FUNC_TRUE];
+              negation = true;
+              right = "";
+            }
+
```

Comment:
@denis8703 It seems weird to be constructing a Condition in this method, only to then return a string array which later needs to be interpreted and have a Condition constructed.  Can the check for ^NULL$ in the right expression be done in parseAtomicCondition?